console.go

```diff
@@ -50,7 +50,7 @@ func NewConsoleSocket(path string) (*Socket, error) {
 // NewTempConsoleSocket returns a temp console socket for use with a container
 // On Close(), the socket is deleted
 func NewTempConsoleSocket() (*Socket, error) {
-	dir, err := ioutil.TempDir("", "pty")
+	dir, err := ioutil.TempDir(os.Getenv("XDG_RUNTIME_DIR"), "pty")
```

Comment:
I think for this one, we need to set the sticky bit on the temp dir that we are creating so that it does not get auto pruned after 6hrs when the socket is not updated.  Here is the runc code for this.

```go
	root := "/run/runc"
	if os.Geteuid() != 0 {
		runtimeDir := os.Getenv("XDG_RUNTIME_DIR")
		if runtimeDir != "" {
			root = runtimeDir + "/runc"
			// According to the XDG specification, we need to set anything in
			// XDG_RUNTIME_DIR to have a sticky bit if we don't want it to get
			// auto-pruned.
			if err := os.MkdirAll(root, 0700); err != nil {
				fatal(err)
			}
			if err := os.Chmod(root, 0700|os.ModeSticky); err != nil {
				fatal(err)
			}
		}
}
```